pretyping/evarconv.ml

```diff
@@ -176,6 +176,10 @@ let check_conv_record env sigma (t1,sk1) (t2,sk2) =
         let s = ESorts.kind sigma s in
 	lookup_canonical_conversion
 	  (proji, Sort_cs (family_of_sort s)),[]
+      | Proj (p, c) ->
+        let c2 = Globnames.ConstRef (Projection.constant p) in
+        let sk2 = Stack.append_app [|c|] sk2 in
+        lookup_canonical_conversion (proji, Const_cs c2), sk2
       | _ ->
```

Comment:
Cool. Well, @maximedenes whenever you finish rebasing this PR, it would be great if you could turn warning 4 on.